pandas/io/parsers.py

```diff
@@ -2508,6 +2512,13 @@ def read(self, rows=None):
             content = content[1:]
 
         alldata = self._rows_to_cols(content)
+        if len(columns) != len(alldata) and notna(alldata[len(columns) :]).any():
```

Comment:
Is there a reason we need the `notna` check?

Reply:
In example mentioned in linked issue additional comma was added in one row. I assumed that additional commas are common and hence we might ignore them and don't raise a warning.
I'm using notna to check if data that won't be included contains only NaN values.